### PR TITLE
feat: Added option to set gradient backgrounds

### LINKED ISSUE #201: New feature - gradient background & text shadow
hello! @DenverCoder1 
   I am suggesting new features like gradient background colors and text-shadow for streak stats card, as shown in below given prototype images. 

## Prototypes with gradient-background
  prototype-1
  prototype-2
  prototype-3

## Prototypes with text-shadow
   prototype-4
   prototype-5
   prototype-6

I think these two new changes will not only provide more aesthetic value to the streak stat card but will also encourage theme contributors to play with color combinations to come up with unique style themes with cool colors. 

Thanks & Regards.

### PR DESCRIPTION
**Pull request by AndroiableDroid**
## Description

Fixes #201 - Half

### Type of change

- [ ] Bug fix (added a non-breaking change which fixes an issue)
- [x] New feature (added a non-breaking change which adds functionality)
- [x] Updated documentation (updated the readme, templates, or other repo files)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

- [x] Tested locally with a valid username
- [x] Tested locally with an invalid username
- [x] Ran tests with `composer test`
- [x] Added or updated test cases to test new features

## Checklist:

- [x] I have checked to make sure no other [pull requests](https://github.com/DenverCoder1/github-readme-streak-stats/pulls?q=is%3Apr+sort%3Aupdated-desc+) are open for this issue
- [x] The code is properly formatted and is consistent with the existing code style
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Screenshots

image

### CODE REVIEW

File: src/card.php [hunk 1]
```diff
@@ -325,6 +329,12 @@ function generateCard(array $stats, array $params = null): string
                 100% { opacity: 1; }
             }
         </style>
+        <defs>
+          <linearGradient id='gradient' gradientTransform='rotate({$gradient[0]})'>
+            <stop offset='0%'  stop-color='#{$gradient[1]}' />
+            <stop offset='100%' stop-color='#{$gradient[2]}' />
+          </linearGradient>
+        </defs>
```
Comment (DenverCoder1) [inline src/card.php:337]: Maybe multiple stops should be supported similar to github-readme-stats?

> image

Comment (DenverCoder1) [reply src/card.php:337]: Not everything needs to be supported by the demo site, it's fine in most cases if the site has less flexibility than directly modifying the parameters directly does

**Comment by DenverCoder1** [comment]
The backend looks great! Thanks for the continued progress :+1:

Is it fine with you if I add some commits to fix some issues with the demo site? 
(some old features are no longer working as they should)

**Comment by AndroiableDroid** [comment]
> The backend looks great! Thanks for the continued progress 👍
> 
> Is it fine with you if I add some commits to fix some issues with the demo site? (some old features are no longer working as they should)

Ya why not, which features?

**Comment by AndroiableDroid** [comment]
> Here are some things I noticed
> 
> * When choosing a color property to add, it seems to add the wrong property (not the one selected)
> * After adding background, it seems that changing other fields no longer updates the image
> * If fields are set to the default value, they can be removed from the url to make it shorter but it seems like that doesn't happen
> * JSON format shows a broken image instead of a codeblock
> * The "Export to PHP" doesn't work with background because of the new format
> * Maybe there should be an option to change the background as a solid color instead of a gradient?
> 
> I'm refactoring some stuff and I'll probably fix most of these along the way. 👍

Do I need to see them or you are handling?

**Comment by AndroiableDroid** [comment]
Looks great.

**Comment by DenverCoder1** [comment]
Everything looks like it's pretty good right now.

Only thing left I can think to consider is whether we should allow users to still set background colors on the demo site without controlling both a start and end color. For example, maybe there could be a radio input to select between solid color and gradient for the background. Possibly it could be two separate menu items, "background (solid)" and "background (gradient)", and adding one would disable the other from being chosen. I'm not entirely sure what's best.

What do you think about this?

**Review by DenverCoder1 — APPROVED** [review]
Looks great! Thanks for the contribution :tada:

**Pull request merged**